### DISCUSSION TITLE
HiDPI support

### DISCUSSION THREAD

**Discussion by jpage4500** [shannah/jdeploy#150] [Q&A]
This was an issue reported to me by someone -- the same person who wasn't able to install the [deb file](https://github.com/shannah/jdeploy/discussions/149).

He was able to install the .bin file but says the app doesn't respect HiDPI settings. I'm not familiar with this and don't have a working Ubuntu environment to even test it.. but, I figured I'd pass it along in case it's something interesting to you:

(ADM = Android Device Manager - the app I'm trying to deploy)

```
This is a common issue with Java apps running on Linux, but on Ubuntu Linux 24.04, installed from the jdeploy release, ADM does not respect HiDPI settings. This means that the window and text are very small on a 4k monitor.



https://wiki.archlinux.org/title/HiDPI 





https://github.com/JetBrains/JetBrainsRuntime 



It looks like JDeploy bundles its own JRE, so even if the system JRE is properly configured, the JDeploy app won’t use it.

``` 

I don't know if just using an existing JRE would simply fix this or not.. but, it is an interesting question.. if a JRE is already installed and meets the criteria (ie: 17+) would it be possible for jdeploy to just use it instead of downloading one?

```
    "jdeploy": {
        "jdk": false,
        "javaVersion": "17",
```

**Comment by shannah** [comment]
This appears to be related to this unresolved bug in the JDK
https://bugs.openjdk.org/browse/JDK-8260270

This gist offers some tips to try to work around it. 
https://gist.github.com/avoidik/9b5ac94a24138ede957789e47d0e381f
 I experimented with the Java one a little and it helped a little bit, but not perfectly.

**Comment by shannah** [comment]
Created ticket #155 .  Closing discussion.
